### ISSUE TITLE
[clang-tidy] cppcoreguidelines-special-member-functions.AllowImplicitlyDeletedCopyOrMove has no effect

### ISSUE THREAD

**Issue by rdong8** [llvm/llvm-project#196615]
The [provided example](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/special-member-functions.html#cmdoption-arg-AllowImplicitlyDeletedCopyOrMove) is still flagged when `AllowImplicitlyDeletedCopyOrMove = true`:

```cpp
#include <boost/core/noncopyable.hpp>

struct A : boost::noncopyable {
  ~A() { std::cout << "dtor\n"; }
};
```

Class 'A' defines a non-default destructor but does not define a move constructor or a move assignment operator
clang-tidy [cppcoreguidelines-special-member-functions](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/special-member-functions.html)

I tried this with clang 19-22 installed via homebrew and Boost 1.90.0 installed on Fedora 44 via dnf and ran into the same issue on all of them.
